### PR TITLE
Switch to registries.conf.d for static binary bundle

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now use a v2 config as well as the `registries.conf.d` location for the configuration file.

#### Which issue(s) this PR fixes:

Refers to https://github.com/cri-o/packaging/pull/17
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Switch to `registries.conf.d` v2 format for static binary bundle.
```

### CODE REVIEW

**Review by haircommander — APPROVED** [review]
/lgtm

Should we move this stuff into the packaging repo so it's closer to the code that calls it?

**Comment by saschagrunert** [comment]
> /lgtm
> 
> Should we move this stuff into the packaging repo so it's closer to the code that calls it?

That could be a long term effort, yes. Where would we split the responsibilities between the repos? For exaxmple, we build the binaries in cri-o/cri-o and just push them to the bucket? cri-o/packaging then uses the binaries and assembles a bundle, pushes it to the bucket and builds packages from it?

**Comment by haircommander** [comment]
> ?

yeah this makes sense to me

File: contrib/bundle/crio-registries.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+unqualified-search-registries = ["docker.io", "quay.io"]
```
Comment (mtrmac) [inline contrib/bundle/crio-registries.conf:1]: A drive-by comment on a PR which, I think, doesn’t change anything in this area:

Shipping two registries, both of which allow registration by ~untrusted users, in the default configuration, is a security risk: https://www.redhat.com/en/blog/be-careful-when-pulling-images-short-name

Comment (saschagrunert) [reply contrib/bundle/crio-registries.conf:1]: Thanks! We added this for convenience, but can reconsider once I'm done with https://github.com/cri-o/cri-o/issues/7380
